eventhubs/client/README.md

```diff
@@ -0,0 +1,347 @@
+
+```java
+EventHubProducer producer = client.createProducer();
+
+// We will publish three events based on simple sentences.
+Flux<EventData> data = Flux.just(
+    new EventData("EventData Sample 1".getBytes(UTF_8)),
+    new EventData("EventData Sample 2".getBytes(UTF_8)),
+    new EventData("EventData Sample 3".getBytes(UTF_8)));
+
+// Send that event. This call returns a Mono<Void>, which we subscribe to. It completes successfully when the
+// event has been delivered to the Event Hub. It completes with an error if an exception occurred while sending
+// the event.
+producer.send(data).subscribe(
+    (ignored) -> System.out.println("Event sent."),
+    error -> {
+        System.err.println("There was an error sending the event: " + error.toString());
+
+        if (error instanceof AmqpException) {
+            AmqpException amqpException = (AmqpException) error;
+
+            System.err.println(String.format("Is send operation retriable? %s. Error condition: %s",
+                amqpException.isTransient(), amqpException.getErrorCondition()));
+        }
+    }, () -> {
+        // Disposing of our producer and client.
+        try {
+            producer.close();
+        } catch (IOException e) {
+            System.err.println("Error encountered while closing producer: " + e.toString());
+            client.close();
+        }
+    }
+);
+```
+
+To send events to a particular partition, set the optional parameter `partitionId` on the `EventHubProducerOptions` 
+when creating an event producer. 
+
+#### Partition ID
+
+```java
+ EventHubProducerOptions producerOptions = new EventHubProducerOptions().partitionId(partitionId);
+ EventHubProducer producer = client.createProducer(producerOptions);
+```
+Many Event Hub operations take place within the scope of a specific partition. Because partitions are owned by the 
+Event Hub, their names are assigned at the time of creation. To understand what partitions are available, You can use 
+the `getPartitionIds` function to get the ids of all available partitions in your Event Hub instance.
+
+```java
+private static final Duration OPERATION_TIMEOUT = Duration.ofSeconds(30);
+...
+
+// We take the first partition id.
+// .blockFirst() here is used to synchronously block until the first partition id is emitted. The maximum wait
+// time is set by passing in the OPERATION_TIMEOUT value. If no item is emitted before the timeout elapses, a
+// TimeoutException is thrown.
+
+String firstPartition = client.getPartitionIds().blockFirst(OPERATION_TIMEOUT);
+```
+#### Partition Key
+
+When an Event Hub producer is not associated with any specific partition, it may be desirable to request that the Event
+ Hubs service keep different events or batches of events together on the same partition. This can be accomplished by 
+ setting a `partition key` when publishing the events.
+ 
+```java
+// The partition key is NOT the identifier of a specific partition. Rather, it is an arbitrary piece of string data
```

Comment:
same with this. these should be "snippets". compile and look at some of the javadocs